### ISSUE TITLE
All text is marked between the text and mouse cursor position

### ISSUE THREAD

**Issue by jakobjakobson13** [flathub/org.tug.texworks#1]
Dear Stefan,

my texworks marks all the text between the mouse cursor and the blinking cursor. 

As I guess this description is a bit short, here comes an description with pictures. If I write the text "Das ist ein Test." with the keyboard without moving the mouse, everything looks fine. And let's call that position where the cursor blinks the "text cursor".
![Bildschirmfoto vom 2020-03-16 18-59-15](https://user-images.githubusercontent.com/43045863/76786959-6dae3c00-67b8-11ea-93bc-8145a094cb62.png)

If I move the mouse, however, all the text between the mouse cursor and the text cursor gets marked as given in the picture, even though I did not click.  
![Bildschirmfoto vom 2020-03-16 18-59-34](https://user-images.githubusercontent.com/43045863/76786958-6d15a580-67b8-11ea-800f-fd079115d988.png)

Perhaps this could be some laptop/trackpad issue, so feel free to ask for more information.

Regards and thanks for your great work.

Jakob

**Comment by stloeffler** [comment]
Unfortunately, I can't reproduce this problem on my machine (Ubuntu 19.10, flatpack 1.4.3)

* Does this problem persist (e.g., after clicking once)?
* Does this also happen if you load a (longer) document and then move the mouse?
* Do you launch the app from the command line?
* Could you run a utility like `evtest` to see if any mouse events out of the ordinary are triggered?

**Comment by jakobjakobson13** [comment]
> Unfortunately, I can't reproduce this problem on my machine (Ubuntu 19.10, flatpack 1.4.3)
> 
I'm running Ubuntu 18.04.4 with Flatpak 1.6.2.

>     * Does this problem persist (e.g., after clicking once)?
> 
It gets activated with the first click in the text window and then persists till you close the window again.

>     * Does this also happen if you load a (longer) document and then move the mouse?
> 
I tried longer documents, this does not seem to make a difference.

>     * Do you launch the app from the command line?
> 
I tried both ways, the graphical and the command line way. Makes no difference and the terminal prints only the following:
```
flatpak run org.tug.texworks 
Failed to create wl_display (No such file or directory)
qt.qpa.plugin: Could not load the Qt platform plugin "wayland" in "" even though it was found.
Gtk-Message: 13:35:30.584: Failed to load module "canberra-gtk-module"
Gtk-Message: 13:35:30.584: Failed to load module "canberra-gtk-module"
```

>     * Could you run a utility like `evtest` to see if any mouse events out of the ordinary are triggered?
I could not see anything unusual.  It was also independent of enabling or disabling the touchpad of my laptop or using an external bluetooth mouse.

**Comment by stloeffler** [comment]
> > * Does this problem persist (e.g., after clicking once)?

> It gets activated with the first click in the text window and then persists till you close the window again.

I'm confused. In your original post, you write (emphasis by me)
> If I move the mouse, however, all the text between the mouse cursor and the text cursor gets marked as given in the picture, *even though I did not click*.

Did you click or not? And if so, where and when? How long? With which mouse button? Did you Alt+Tab (or similar) at any point?

**Comment by jakobjakobson13** [comment]
Okay, I admit the previous posts were a bit confusing, so here's a longer explanation.

By starting texworks via the terminal you can directly start typing without setting the cursor by a mouse click. That means no mouse click has been done and so far texworks works as supposed and does not mark the text between the mouse cursor and the text cursor.

If I click (left mouse button click) somewhere in the text editor field, the marking between the text and mouse cursor position applies. So, the marking starts with the first click. The marking due to the first click is independent of the text length,  the number of the following mouse clicks (left mouse button click), a window change to firefox for example or opening the texworks preferences.

I hope that was a bit clearer, otherwise let me know.

**Comment by stloeffler** [comment]
It seems this is caused by a recent change in Qt. I have committed a change to the latest development version that should hopefully fix this problem, but the flathub package will likely only be updated with the next stable release. It would be very helpful to figure out if the change did what it is supposed to do. To get the latest development version to test, you could
* Build from [the source](https://github.com/TeXworks/texworks)
* Use the ["Latest builds" ppa on Launchpad](https://launchpad.net/~texworks/+archive/ubuntu/ppa)

**Comment by gcabanac** [comment]
I compiled the GIT version (with QT 5.14.1 on Mac OS 10.15.3) and the observed problem is fixed. Thank you!

**Comment by stloeffler** [comment]
This should be fixed now with the 0.6.5 release.
